### ISSUE TITLE
spaceAroundParens adds space between [] and () when constructing arrays

### ISSUE THREAD

**Issue by tobihagemann** [nicklockwood/SwiftFormat#830]
Not sure how to describe it more properly but consider this code:

```swift
return withUnsafeBytes(of: self, { [UInt8]($0) })
```

SwiftFormat wants to add a space in `[UInt8]($0)` so that it becomes `[UInt8] ($0)`. I don't think that this is correct because other code that constructs an array is not affected:

```swift
var buffer = [UInt8](repeating: 0x00, count: maxLength)
```

I'm pretty sure that this behavior was introduced in some update. I'm running SwiftFormat 0.47.8 now but I don't remember which version didn't cause this.

**Comment by nicklockwood** [comment]
@tobihagemann this is probably due to it mistaking the [...] for a closure capture list. I believe it was introduced in 0.47.8

**Comment by nicklockwood** [comment]
@tobihagemann fixed in 0.47.9.
